ecosystem/typescript/sdk/src/plugins/aptos_token.ts

```diff
@@ -0,0 +1,490 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable max-len */
+
+import { AptosAccount } from "../account/aptos_account";
+import { AnyNumber } from "../bcs";
+import { MAX_U64_BIG_INT } from "../bcs/consts";
+import { Provider } from "../providers";
+import { AptosClient, OptionalTransactionArgs } from "../providers/aptos_client";
+import { TransactionBuilderRemoteABI } from "../transaction_builder";
+import { HexString, MaybeHexString } from "../utils";
+import { getPropertyValueRaw, getSinglePropertyValueRaw } from "../utils/property_map_serde";
+
+export interface CreateCollectionOptions {
+  royaltyNumerator?: number;
+  royaltyDenominator?: number;
+  mutableDescription?: boolean;
+  mutableRoyalty?: boolean;
+  mutableURI?: boolean;
+  mutableTokenDescription?: boolean;
+  mutableTokenName?: boolean;
+  mutableTokenProperties?: boolean;
+  mutableTokenURI?: boolean;
+  tokensBurnableByCreator?: boolean;
+  tokensFreezableByCreator?: boolean;
+}
+
+const PropertyTypeMap = {
+  BOOLEAN: "bool",
+  U8: "u8",
+  U16: "u16",
+  U32: "u32",
+  U64: "u64",
+  U128: "u128",
+  U256: "u256",
+  ADDRESS: "address",
+  VECTOR: "vector<u8>",
+  STRING: "string",
+};
+
+export type PropertyType = keyof typeof PropertyTypeMap;
+
+/**
+ * Class for managing aptos_token
+ */
+export class AptosToken {
+  readonly provider: Provider;
+
+  private readonly tokenType: string = "0x4::token::Token";
+
+  /**
+   * Creates new AptosToken instance
+   *
+   * @param provider Provider instance
+   */
+  constructor(provider: Provider) {
+    this.provider = provider;
+  }
+
+  private async submitTransaction(
+    account: AptosAccount,
+    funcName: string,
+    typeArgs: string[],
+    args: any[],
+    extraArgs?: OptionalTransactionArgs,
+  ) {
+    const builder = new TransactionBuilderRemoteABI(this.provider, {
```

Comment:
Hmm this feels like something that can be abstracted into a common func?

Reply:
this is already a common func in this file, a PR to make it a global func in the SDK will be done in a future and separate work